api/src/routes/edit.ts

```diff
@@ -1,10 +1,59 @@
 import express = require("express");
+import bodyParser = require("body-parser");
 import Config from "../models/Config";
-const router = express.Router();
+import Fedora from "../services/Fedora";
+import FedoraObjectFactory from "../services/FedoraObjectFactory";
+import MetadataExtractor from "../services/MetadataExtractor";
 import { requireToken } from "./auth";
 
+const router = express.Router();
+
 router.get("/models", requireToken, function (req, res) {
     res.json({ CollectionModels: Config.getInstance().collectionModels, DataModels: Config.getInstance().dataModels });
 });
 
+router.post("/object/new", requireToken, bodyParser.json(), async function (req, res) {
+    let parentPid = req.body.parent ?? null;
```

Comment:
Is this an empty string? You can try null coalescing operator req?.body?.parent;

Reply:
Oh I meant optional chaining. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining

Reply:
My goal is to normalize to null if it's an empty string, and keep the string otherwise. I'm used to PHP-style null coalescing, but I see that I would do better to use Javascript [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) here. Thanks!

Reply:
Pushed up an improvement; I'm sure there are other places in the code where we can simplify some ugly code with this technique.